### PR TITLE
ADD User to MISP

### PR DESCRIPTION
**Pull request by yilmazdincer**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
I want to add a new user to misp, so I created a function called add_user_to_misp in the MISP integration.
Added three command:
-> misp-add-user
-> misp-get-organization-info
-> misp-get-role-info

## Screenshots
image

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 1]
```diff
@@ -2486,11 +2486,31 @@ script:
...
+      type: textArea
+    - description: ID number indicating which organization the new user will be added to.
+      name: org_id
+      required: true
+      type: textArea
+    - description: Role of the new user to be added
+      name: role_id
+      required: true
+      type: textArea
+    - description: A password for the new user.
```
Comment (RotemAmit) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2502]: Please add to the description the password requirements (length, numbers or letters etc.).

**Review by RotemAmit — COMMENTED** [review]
Thank you very much for your contribution! Nice work!
Here are some notes:
* After you are updating the YML file with the outputs, as in the comments below, please update the README file accordingly. Here is a [link](https://xsoar.pan.dev/docs/documentation/readme_file) to an explanation about the README file.
* Please add unit tests to the commands you added. You can look at the existing tests in the file Packs/MISP/Integrations/MISPV3/MISPV3_test.py and for more information there is a [link](Packs/MISP/Integrations/MISPV3/MISPV3_test.py) with an explanation about writing unit tests.
* Please add the new commands to the test playbook in the file Packs/MISP/TestPlaybooks/MISP_V3_Test.yml . In order to to this please upload this file to XSOAR, then navigate to the Playbooks page, and search for: MISP V3 Test. Edit the playbook and add the new commands. Then download it add replace the old test playbook with the updated one (use the same file name and location).
* Please fix the validations errors. In the step 'ci/circleci: Run Validations' click on details, in there you will see the list of the validations errors. 

If you have any question feel free to ask me here or via Slack.

**Review by RotemAmit — COMMENTED** [review]
Nice work!
I reviewed the content in the PR and would like to provide some feedback:
* Please run 'demisto-sdk pre-commit -i Packs/MISP/Integrations/MISPV3/MISPV3.py'. This command will do some changed to the code, please commit them and push it to the PR.
* Please add the new commands to the test playbook.
* Please add to the new test functions a doc string that will describe the unit test scenario. Given <some input>, When <some state> Then <ensure some behviour>. For examples you can check out the rest of the unit tests of the integration.
Thank you for your contribution!

File: Packs/MISP/Integrations/MISPV3/MISPV3.py [hunk 2]
```diff
@@ -537,6 +536,71 @@ def create_event_command(demisto_args: dict):
...
+    new_user.email = demisto_args.get('email')
+    new_user.org_id = demisto_args.get('org_id')
+    new_user.role_id = demisto_args.get('role_id')
+    new_user.password = demisto_args.get('password')
+    response = PYMISP.add_user(new_user)
+    if 'errors' in response:
+        raise DemistoException(f'Failed to add user.\nError message: {response}')
+    else:
+        human_readable = f"## MISP add user\nNew user was added to MISP.\nEmail:{new_user.email}"
+        return CommandResults(readable_output=human_readable, raw_response=response, outputs=response)
```
Comment (RotemAmit) [inline Packs/MISP/Integrations/MISPV3/MISPV3.py:555]:
```suggestion
        return CommandResults(readable_output=human_readable, raw_response=response, outputs=response, outputs_prefix='MISP.Organization')
```
Please add the outputs_prefix param to the CommandResults in order that the output will match MISP.User.id

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 3]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+    name: misp-add-user
+    outputs:
+    - contextPath: MISP.User.id
+      description: MISP User ID.
+      type: string
+    - contextPath: MISP.User.password
+      description: MISP User Password.
+      type: string
+    - contextPath: MISP.User.org_id
+      description: MISP User Organisation ID.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2509]:
```suggestion
      description: MISP user organization ID.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 4]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.server_id
+      description: MISP User Server ID.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2512]:
```suggestion
      description: MISP user server ID.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 5]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.email
+      description: MISP User Email.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2515]:
```suggestion
      description: MISP user email.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 6]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.autoalert
+      description: MISP User Auto Alert.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2518]:
```suggestion
      description: MISP user auto alert.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 7]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: boolean
+    - contextPath: MISP.User.authkey
+      description: MISP User Auth Key.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2521]:
```suggestion
      description: MISP user auth key.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 8]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.invited_by
+      description: MISP User Invited By.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2524]:
```suggestion
      description: MISP user invited by.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 9]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.gpgkey
+      description: MISP User GPG Key.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2527]:
```suggestion
      description: MISP user GPG key.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 10]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.certif_public
+      description: MISP User Certif Public.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2530]:
```suggestion
      description: MISP user public certificate.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 11]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.nids_sid
+      description: MISP User Nids Sid.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2533]:
```suggestion
      description: MISP user Network Intrusion Detection System (NIDS) Signature ID (SID).
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 12]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.termsaccepted
+      description: MISP User Terms Accepted.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2536]:
```suggestion
      description: Whether MISP user terms were accepted.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 13]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: boolean
+    - contextPath: MISP.User.newsread
+      description: MISP User News Read.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2539]:
```suggestion
      description: MISP user news read.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 14]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.role_id
+      description: MISP User Role ID.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2542]:
```suggestion
      description: MISP user role ID.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 15]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.change_pw
+      description: MISP User Change Password.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2545]:
```suggestion
      description: Whether the MISP user password was changed.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 16]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: boolean
+    - contextPath: MISP.User.contactalert
+      description: MISP User Contact Alert.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2548]:
```suggestion
      description: MISP user contact alert.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 17]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: boolean
+    - contextPath: MISP.User.disabled
+      description: MISP User Disabled.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2551]:
```suggestion
      description: Whether the MISP user was disabled.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml (hunk 16 above)
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2548]: Not really sure what this description means.  The value is a boolean.

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 18]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: boolean
+    - contextPath: MISP.User.expiration
+      description: MISP User Expiration.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2554]:
```suggestion
      description: MISP user expiration.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 19]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.current_login
+      description: MISP User Current Login.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2557]:
```suggestion
      description: MISP user current login.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 20]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.last_login
+      description: MISP User Last Login.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2560]:
```suggestion
      description: MISP user last login.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 21]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.force_logout
+      description: MISP User Force Logout.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2563]:
```suggestion
      description: MISP user force logout.
```
Should this be "Whether the MISP user was forced to log out."

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 22]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: boolean
+    - contextPath: MISP.User.date_created
+      description: MISP User Created Date.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2566]:
```suggestion
      description: MISP user created date.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 23]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.User.date_modified
+      description: MISP User Modified Date.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2569]:
```suggestion
      description: MISP user modified date.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 24]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+  - arguments: []
+    description: Display organization ids and names.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2572]:
```suggestion
    description: Display the organization IDs and names.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 25]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+    name: misp-get-organization-info
+    outputs:
+    - contextPath: MISP.Organization.id
+      description: MISP Ogranization ID.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2576]:
```suggestion
      description: MISP organization ID.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 26]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.Organization.name
+      description: MISP Organization Name.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2579]:
```suggestion
      description: MISP organization name.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 27]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+  - arguments: []
+    description: Display role names and role ids.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2582]:
```suggestion
    description: Display role names and role IDs.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 28]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+    name: misp-get-role-info
+    outputs:
+    - contextPath: MISP.Role.id
+      description: MISP Role ID.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2586]:
```suggestion
      description: MISP role ID.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 29]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      type: string
+    - contextPath: MISP.Role.name
+      description: MISP Role Name.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2589]:
```suggestion
      description: MISP role name.
```

File: Packs/MISP/Integrations/MISPV3/MISPV3.yml [hunk 30]
```diff
@@ -2480,6 +2480,114 @@ script:
...
+      required: true
+      type: textArea
+    - description: Role of the new user to be added
+      name: role_id
+      required: true
+      type: textArea
+    - description: A password for the new user. Ensure that the password is at least 12 characters long, contains at least one upper-case, includes a digit or a special character, at least one lower-case character.
+      name: password
+      required: true
+    description: Adding a new user to MISP.
```
Comment (ShirleyDenkberg) [inline Packs/MISP/Integrations/MISPV3/MISPV3.yml:2499]:
```suggestion
    description: Add a new user to MISP.
```

File: Packs/MISP/ReleaseNotes/2_1_28.md [hunk 31]
```diff
@@ -0,0 +1,8 @@
+#### Integrations
+
+##### MISP v3
+
+- Added the following commands:
+    - ***misp-add-user***
```
Comment (ShirleyDenkberg) [inline Packs/MISP/ReleaseNotes/2_1_28.md:6]:
```suggestion
- ***misp-add-user***

```

File: Packs/MISP/ReleaseNotes/2_1_28.md [hunk 32]
```diff
@@ -0,0 +1,8 @@
+#### Integrations
+
+##### MISP v3
+
+- Added the following commands:
+    - ***misp-add-user***
+    - ***misp-get-organization-info***
```
Comment (ShirleyDenkberg) [inline Packs/MISP/ReleaseNotes/2_1_28.md:7]:
```suggestion
- ***misp-get-organization-info***

```

File: Packs/MISP/ReleaseNotes/2_1_28.md [hunk 33]
```diff
@@ -0,0 +1,8 @@
+#### Integrations
+
+##### MISP v3
+
+- Added the following commands:
+    - ***misp-add-user***
+    - ***misp-get-organization-info***
+    - ***misp-get-role-info***
```
Comment (ShirleyDenkberg) [inline Packs/MISP/ReleaseNotes/2_1_28.md:8]:
```suggestion
- ***misp-get-role-info***
```

File: Packs/MISP/ReleaseNotes/2_1_28.md [hunk 34]
```diff
@@ -0,0 +1,8 @@
+#### Integrations
+
+##### MISP v3
+
+- Added the following commands:
```
Comment (ShirleyDenkberg) [inline Packs/MISP/ReleaseNotes/2_1_28.md:5]:
```suggestion
Added the following commands:

```

**Comment by ShirleyDenkberg** [comment]
@yilmazdincer After implementing my comments, please regenerate the integration readme.

@RotemAmit Doc review completed.

**Review by RotemAmit — APPROVED** [review]
Thank you very much for your contribution!
Your hard work and dedication is highly appreciated!

**Pull request merged**